dms/models/directory.py

```diff
@@ -273,13 +282,11 @@ def check_access(self, operation, raise_exception=False):
     )
     storage_id_save_type = fields.Selection(related="storage_id.save_type", store=False)
 
-    @api.depends("root_storage_id", "storage_id")
+    @api.depends("storage_id")
     def _compute_allowed_model_ids(self):
         for record in self:
             record.allowed_model_ids = False
-            if record.root_storage_id and record.root_storage_id.model_ids:
-                record.allowed_model_ids = record.root_storage_id.model_ids.ids
-            elif record.storage_id and record.storage_id.model_ids:
+            if record.storage_id and record.storage_id.model_ids:
```

Comment:
No need of this double condition:
```suggestion
            if record.storage_id.model_ids:
```
If `record.storage_id` is an empty recordset, any subfield with dot notation will be populated as empty.

Reply:
Even more, why don't you assign directly without if?:
```python
        record.allowed_model_ids = record.storage_id.model_ids.ids
```